modules/gcp/compute.go

```diff
@@ -224,6 +225,87 @@ func (i *Instance) SetLabelsE(t *testing.T, labels map[string]string) error {
 	return nil
 }
 
+// GetMetadata gets the given Compute Instance's metadata
+func (i *Instance) GetMetadata(t *testing.T) []*compute.MetadataItems {
+	return i.Metadata.Items
+}
+
+// SetMetadata sets the given Compute Instance's metadata
+func (i *Instance) SetMetadata(t *testing.T, metadata map[string]string) {
+	err := i.SetMetadataE(t, metadata)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// SetLabelsE adds the given metadata map to the existing metadata of the given Compute Instance.
+func (i *Instance) SetMetadataE(t *testing.T, metadata map[string]string) error {
+	logger.Logf(t, "Adding metadata to instance %s in zone %s", i.Name, i.Zone)
+
+	ctx := context.Background()
+	service, err := NewInstancesServiceE(t)
+	if err != nil {
+		return err
+	}
+
+	metadataItems := newMetadata(t, i.Metadata, metadata)
+	req := service.SetMetadata(i.projectID, i.GetZone(t), i.Name, metadataItems)
+	if _, err := req.Context(ctx).Do(); err != nil {
+		return fmt.Errorf("Instances.SetMetadata(%s) got error: %v", i.Name, err)
+	}
+
+	return nil
+}
+
+// newMetadata takes in a Compute Instance's existing metadata plus a new set of key-value pairs and returns an updated
+// metadata object.
+func newMetadata(t *testing.T, oldMetadata *compute.Metadata, kvs map[string]string) *compute.Metadata {
+	items := []*compute.MetadataItems{}
+
+	for key, val := range kvs {
+		item := &compute.MetadataItems{
+			Key:   key,
+			Value: &val,
+		}
+
+		items = append(oldMetadata.Items, item)
+	}
+
+	newMetadata := &compute.Metadata{
+		Fingerprint: oldMetadata.Fingerprint,
+		Items:       items,
+	}
+
+	return newMetadata
+}
+
+// Add the given public SSH key to the Compute Instance. Users can SSH in with the given username.
+func (i *Instance) AddSshKey(t *testing.T, username string, publicKey string) {
+	err := i.AddSshKeyE(t, username, publicKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// Add the given public SSH key to the Compute Instance. Users can SSH in with the given username.
+func (i *Instance) AddSshKeyE(t *testing.T, username string, publicKey string) error {
+	logger.Logf(t, "Adding SSH Key to Compute Instance %s for username %s\n", i.Name, username)
+
+	publicKeyFormatted := strings.TrimSpace(publicKey)
+	sshKeyFormatted := fmt.Sprintf("%s:%s %s", username, publicKeyFormatted, username)
```

Comment:
Seems very weird to have to do this formatting manually... Is this some GCP thing?

Reply:
It is indeed a GCP-specific format. I added a comment to clarify and linked to the [relevant docs](https://cloud.google.com/compute/docs/instances/adding-removing-ssh-keys).